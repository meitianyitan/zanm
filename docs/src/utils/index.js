function iframeReady(iframe, callback) {
  const doc = iframe.contentDocument || iframe.contentWindow.document;
  const interval = () => {
    if (iframe.contentWindow.changePath) {
      callback();
    } else {
      setTimeout(() => {
        interval();
      }, 50);
    }
  };

  if (doc.readyState === 'complete') {
    interval();
  } else {
    iframe.onload = interval;
  }
}

const ua = navigator.userAgent.toLowerCase();
const isMobile = /ios|iphone|ipod|ipad|android/.test(ua);

export {
  isMobile,
  iframeReady
};
