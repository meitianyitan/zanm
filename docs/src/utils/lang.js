import Locale from '../../../packages/locale';
import zhCN from '../../../packages/locale/lang/zh-CN';
import enUS from '../../../packages/locale/lang/en-US';

const langMap = {
  'en-US': {
    title: 'Zanm - A Mobile UI for Vue.js 2.0',
    messages: enUS
  },
  'zh_CN': {
    title: 'Zanm - 一套基于 Vue.js 的移动端组件库',
    messages: zhCN
  }
};
let currentLang = '';

setLang(getDefaultLang());

function getDefaultLang() {
  const langs = Object.keys(langMap);
  const hash = location.hash;

  for (let i = 0; i < langs.length; i++) {
    if (~hash.indexOf(langs[i])) {
      return langs[i];
    }
  }

  const userLang = localStorage.getItem('ZANM_LANGUAGE') || navigator.language || 'en_US';
  return ~userLang.indexOf('zh-') ? 'zh_CN' : 'en_US';
}

export function setLang(lang) {
  if (currentLang === lang) {
    return;
  }

  currentLang = lang;
  if (window.localStorage) {
    localStorage.setItem('ZANM_LANGUAGE', lang);
  }
  Locale.use(lang, langMap[lang].messages);
  document.title = langMap[lang].title;
}
