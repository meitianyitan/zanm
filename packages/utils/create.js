/**
 * Create a component with common options
 */
import createBasic from './create-basic';
import Loading from '../loading';

export default function(sfc) {
  sfc.components = Object.assign(sfc.components || {}, {
    Loading
  });
  return createBasic(sfc);
};
