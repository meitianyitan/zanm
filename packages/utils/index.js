function isDef(value) {
  return value !== undefined && value !== null;
}

function isObj(x) {
  const type = typeof x;
  return x !== null && (type === 'object' || type === 'function');
}

function get(object, path) {
  const keys = path.split('.');
  let result = object;

  keys.forEach(key => {
    result = isDef(result[key]) ? result[key] : '';
  });

  return result;
}

const camelizeRE = /-(\w)/g;
function camelize(str) {
  return str.replace(camelizeRE, (_, c) => c.toUpperCase());
}

export {
  isDef,
  isObj,
  get,
  camelize
};
