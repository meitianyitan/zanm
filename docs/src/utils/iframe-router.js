/**
 * 同步父窗口和 iframe 的 vue-router 状态
 */
import { setLang } from './lang';
import { iframeReady, isMobile } from './';

window.syncPath = function(dir) {
  const router = window.vueRouter;
  const isInIframe = window !== window.top;
  const currentDir = router.history.current.path;
  const pathParts = currentDir.split('/');
  let lang = pathParts[0];
  if (currentDir[0] === '/') {
    lang = pathParts[1];
  }

  if (!isInIframe && !isMobile) {
    const iframe = document.querySelector('iframe');
    if (iframe) {
      iframeReady(iframe, () => {
        iframe.contentWindow.changePath(lang, currentDir);
      });
    }
    setLang(lang);
  } else if (isInIframe) {
    window.top.changePath(lang, currentDir);
  }
};

window.changePath = function(lang, path = '') {
  setLang(lang);
  window.vueRouter.replace(path);
};
