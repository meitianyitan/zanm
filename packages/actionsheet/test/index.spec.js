import { mount } from '../../../test/utils';
import Actionsheet from '../';

test('callback events', () => {
  const callback = jest.fn();
  const wrapper = mount(Actionsheet, {
    propsData: {
      value: true,
      actions: [
        { name: 'Option', callback },
        { name: 'Option', disabled: true }
      ],
      cancelText: 'Cancel'
    }
  });

  const options = wrapper.findAll('li');
  options.at(0).trigger('click');
  options.at(1).trigger('click');
  wrapper.find('.zan-actionsheet__cancel').trigger('click');

  expect(callback.mock.calls.length).toBe(1);
  expect(wrapper.emitted('cancel')).toBeTruthy();
  expect(wrapper.emitted('input')[0][0]).toBeFalsy();
  expect(wrapper.emitted('select')[0][0]).toBeTruthy();
  expect(wrapper.emitted('select')[0][1]).toBeFalsy();
  expect(wrapper).toMatchSnapshot();
});
