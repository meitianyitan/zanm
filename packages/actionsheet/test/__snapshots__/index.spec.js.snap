// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`callback events 1`] = `
<div class="zvm-actionsheet" name="zvm-slide-bottom">
  <ul class="zvm-hairline--bottom">
    <li class="zvm-actionsheet__item zvm-hairline--top"><span class="zvm-actionsheet__name">Option</span>
      <!---->
    </li>
    <li class="zvm-actionsheet__item zvm-actionsheet__item--disabled zvm-hairline--top"><span class="zvm-actionsheet__name">Option</span>
      <!---->
    </li>
  </ul>
  <div class="zvm-actionsheet__cancel zvm-hairline--top">Cancel</div>
</div>
`;
